packages/transactions/src/signer.ts

```diff
@@ -19,6 +19,38 @@ export class TransactionSigner {
     this.originDone = false;
     this.checkOversign = true;
     this.checkOverlap = true;
+
+    // If multi-sig spending condition exists, iterate over
+    // auth fields and reconstruct sigHash
+    let spendingCondition = transaction.auth.spendingCondition;
+    if (spendingCondition && !isSingleSig(spendingCondition)) {
+      if (
+        spendingCondition.fields.filter(
+          field => field.contents.type === StacksMessageType.MessageSignature
+        ).length >= spendingCondition.signaturesRequired
+      ) {
+        throw new Error('SpendingCondition has more signatures than are expected');
+      }
+
+      spendingCondition.fields.forEach(field => {
+        if (field.contents.type === StacksMessageType.MessageSignature) {
+          if (!transaction.auth.authType) {
+            throw Error('"transaction.auth.authType" not defined');
+          }
+
+          const signature = field.contents;
+          const nextVerify = nextVerification(
+            this.sigHash,
+            transaction.auth.authType,
+            spendingCondition!.fee,
+            spendingCondition!.nonce,
```

Comment:
Are these force assertions still needed? If so, should this instead use some undefined checks and explicit errors instead?

Reply:
these are removed in the followup PR that I have open (the Authorization refactor)

Reply:
Ah thats right, thanks!